internal/tiltfile/tiltfile_state.go

```diff
@@ -884,8 +884,22 @@ func (s *tiltfileState) extractEntities(dest *k8sResource, imageRef container.Re
 	return nil
 }
 
+// registry returns the image registry we should use; if detected, a pre-configured
+// local registry; otherwise, the registry specified by the user via default_registry.
+// Otherwise, we'll return the zero value of `s.defaultReg`, which is an empty registry.
+func (s *tiltfileState) registry() container.Registry {
```

Comment:
i feel like this should be called decideRegistry() or smth to make it clear this has side-effects and should only be called once